src/portable/ti/msp430x5xx/dcd_msp430x5xx.c

```diff
@@ -207,6 +204,24 @@ void dcd_remote_wakeup(uint8_t rhport)
   (void) rhport;
 }
 
+void dcd_connect(uint8_t rhport)
+{
+  (void) rhport;
+
+  USBKEYPID = USBKEY;
+  USBCNF |= PUR_EN; // Enable pullup.
+  USBKEYPID = 0;
```

Comment:
Is there a concurrency issue here?

USBKEYPID is also used in ```bus_reset```, which looks to be called from the ISR?

Reply:
ah yeah, we better to be safe, just take a look at dcd_init_enable()/disable() look like there is quite a lot of stuff. We can simply call it, unless @cr1901 know a better way :)